### DISCUSSION TITLE
WIP: Using cobbler to manage DHCP for a local libvirt NAT network.

### DISCUSSION THREAD

**Discussion by opoplawski** [cobbler/cobbler#3510] [Show and tell]
### Overview
I'd like to have cobbler configured to be able to do network installs of VMs on my laptop.  To do this, we need to replace the built-in default network's dnsmasq configuration with one from cobbler.  Here is what I have so far.

### Disable libvirt's default network dnsmasq
Run `virsh net-edit default` and remove any `dhcp` tag such as:
```
    <dhcp>
      <range start='192.168.122.100' end='192.168.122.254'/>
      <host mac='52:54:00:d6:76:d7' name='f37' ip='192.168.122.2'/>
    </dhcp>
```
Then disable dns by adding this to the top level of the `network` tag:
```
  <dns enable='no'/>
```
Then stop and start the network.  NOTE: this will break the networking of any currently running VMs that use this network.
```
virsh net-destroy default
virsh net-start default
```
### Configure cobbler to configure dnsmasq
Edit `/etc/cobbler/settings.yml` and enable dnsmasq:
```
modules:
...
  dhcp:
    module: "managers.dnsmasq"
```
Restart cobblerd:
```
systemctl restart cobblerd
```
Configure cobbler's `/etc/cobbler/dnsmasq.template`:
```
interface=virbr0
dhcp-range=192.168.122.100,192.168.122.254,255.255.255.0
dhcp-no-override
dhcp-authoritative
dhcp-lease-max=155
dhcp-hostsfile=/var/lib/libvirt/dnsmasq/default.hostsfile
addn-hosts=/var/lib/libvirt/dnsmasq/default.addnhosts
addn-hosts = /var/lib/cobbler/cobbler_hosts

dhcp-option=66,$next_server_v4
dhcp-authoritative
dhcp-boot=pxelinux.0
dhcp-boot=net:normalarch,pxelinux.0

$insert_cobbler_system_definitions
```
Sync:
```
cobbler sync
```

**Comment by SchoolGuy** [comment]
I am at the moment also preparing https://github.com/cobbler/cobbler.github.io/pull/178.
